### PR TITLE
Change api of additional files uploading

### PR DESCRIPTION
**Pull request by kgevorkyan**
### What's done:
* Change api of additional files uploading - process low level information at the backend side

### CODE REVIEW

File: save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/FileInfo.kt [hunk 1]
```diff
@@ -16,6 +16,24 @@ data class FileInfo(
...
+        this.isExecutable,
+    )
+}
+
+/**
+ * @property name name of a file
+ * @property isExecutable whether the file is executable
+ */
+@Serializable
+data class FileInfoDto(
```
Comment (petertrr) [inline save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/FileInfo.kt:34]: This name kind of contradicts usual convention Entity/Dto. Maybe, call this class `ShortFileInfo` or rename the original class into `VerboseFileInfo` or `FileInfoWithMetadata`?

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt [hunk 2]
```diff
@@ -736,18 +736,27 @@ class ProjectView : AbstractView<ProjectExecutionRouteProps, ProjectViewState>(f
                 }
 
                 element.files!!.asList().forEach { file ->
-                    val response: FileInfo = post(
+                    val response: FileInfoDto = post(
                         "$apiUrl/files/upload",
```
Comment (kgevorkyan) [inline save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt:740]: Here we can use some flag, which will responsible of returning value `FileInfo` or `FileInfoDto`

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt [hunk 3]
```diff
@@ -736,18 +736,27 @@ class ProjectView : AbstractView<ProjectExecutionRouteProps, ProjectViewState>(f
...
+                        },
+                        Json.encodeToString(response)
+                    ).decodeFromJsonString()
+
                     setState {
                         // add only to selected files so that this entry isn't duplicated
-                        files.add(response)
-                        bytesReceived += response.sizeBytes
+                        files.add(fileInfo)
+                        bytesReceived += fileInfo.sizeBytes
```
[the remark this answers is not in the document]
Comment (kgevorkyan) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt]: We actually have `suiteByteSize` for this (at the beginning of this function), here we specifically collect information about received bytes

File: save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/FileInfo.kt (hunk 1 above)
Comment (kgevorkyan) [reply save-cloud-common/src/commonMain/kotlin/org/cqfn/save/domain/FileInfo.kt]: Renamed to `ShortFileInfo`

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt (hunk 2 above)
Comment (kgevorkyan) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt]: Added

**Review by petertrr — APPROVED** [review]

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**